micro-rdk/src/common/data_manager.rs

```diff
@@ -383,6 +398,10 @@ where
                                 current_chunk_size = next_message.len();
                                 current_chunk = vec![next_message];
                             }
+                            #[cfg(feature = "esp32")]
```

Comment:
doesn't have to be feature gated

Reply:
It does because I had to feature-gate the weak symbol in the first place, but I suppose I could make an alternate plain empty implementation of `data_upload_callback` when the esp32 feature is not active. Is this preferred?

Reply:
make sense